### PR TITLE
risc-v: Let g_cpu_basestack determined at compile time

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary

The cpux's idle stack is loaded from g_cpu_basestack (data section) now, but on this time it maybe not ready since it is initialized by cpu0, then we must wait for g_cpu_basestack ready before use it.

And use textual label in k210_head.S
## Impact
K210 SMP only
## Testing
K210 QEMU and real device.

### CODE REVIEW

**Comment by masayuki2009** [comment]
@no1wudi 
This PR works fine but please 
1. Fixup the 2 commits into 1 commit.
2. And add commit log on fixing the bug reported in https://github.com/apache/incubator-nuttx/pull/5484#issuecomment-1038802121

**Comment by no1wudi** [comment]
> @no1wudi This PR works fine but please
> 
> 1. Fixup the 2 commits into 1 commit.
> 2. And add commit log on fixing the bug reported in [risc-v: Implement common up_cpu_idlestack #5484 (comment)](https://github.com/apache/incubator-nuttx/pull/5484#issuecomment-1038802121)

Commit message updated, although the first commit is very small, it is logically independent, it fixed another potential bug on non-smp case: load a value from overflowed address from g_cpu_basestack.

File: arch/risc-v/src/k210/k210_head.S [hunk 1]
```diff
@@ -48,15 +48,29 @@ __start:
...
   /* Set stack pointer to the idle thread stack */
 
-  bnez a0, 1f
+  bnez a0, boot_slave_core
   la   sp, K210_IDLESTACK0_TOP
-  j    2f
-1:
+  j    common_startup
+
+boot_slave_core:
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply arch/risc-v/src/k210/k210_head.S:55]: Just for better readability

File: arch/risc-v/src/k210/k210_head.S [hunk 2]
```diff
@@ -71,11 +78,6 @@ __start:
   li   t0, CONFIG_IDLETHREAD_STACKSIZE
   add  sp, sp, t0
 
-  /* In case of single CPU config, stop here */
-
-#if !(defined CONFIG_SMP) || (CONFIG_SMP_NCPUS == 1)
-  wfi
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/k210/k210_head.S:77]: why still need change here? if we change the stack pointer array to const

File: arch/risc-v/src/common/riscv_cpuidlestack.c [hunk 3]
```diff
@@ -83,7 +83,7 @@ static uint8_t aligned_data(16) cpu7_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
  * Public Data
  ****************************************************************************/
 
-uint8_t *g_cpu_basestack[CONFIG_SMP_NCPUS] =
+uint8_t * const g_cpu_basestack[CONFIG_SMP_NCPUS] =
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_cpuidlestack.c:86]: change to const uint8_t *

File: arch/risc-v/src/common/riscv_internal.h [hunk 4]
```diff
@@ -117,7 +117,7 @@ EXTERN uintptr_t g_idle_topstack;
 
 /* Address of per-cpu idle stack base */
 
-EXTERN uint8_t *g_cpu_basestack[CONFIG_SMP_NCPUS];
+EXTERN uint8_t * const g_cpu_basestack[CONFIG_SMP_NCPUS];
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_internal.h:120]: ditto

File: arch/risc-v/src/k210/k210_head.S [hunk 5]
```diff
@@ -55,7 +55,7 @@ __start:
 
   /* In case of single CPU config, stop here */
 
-#if CONFIG_SMP_NCPUS == 1
+#if !(defined CONFIG_SMP) || (CONFIG_SMP_NCPUS == 1)
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/k210/k210_head.S:58]: move to previous patch instead

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@masayuki2009 could you try the new approach?

File: arch/risc-v/src/k210/k210_head.S [hunk 6]
```diff
@@ -53,6 +53,13 @@ __start:
   j    2f
 1:
 
+  /* In case of single CPU config, stop here */
+
+#if !(defined CONFIG_SMP) || (CONFIG_SMP_NCPUS == 1)
```
Comment (pkarashchenko) [inline arch/risc-v/src/k210/k210_head.S:58]:
```suggestion
#if !defined(CONFIG_SMP) || (CONFIG_SMP_NCPUS == 1)
```

**Comment by masayuki2009** [comment]
> @masayuki2009 could you try the new approach?

@no1wudi @xiaoxiang781216 
Thanks, the new approach works.

**Review by masayuki2009 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**